### PR TITLE
*: remove TLS1.0, TLS1.1 support

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: ref #36036 

Problem Summary:

### What changed and how does it work?

TLS1.0/1.1 has security issues, remove the support of them.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [X] Manual test (add detailed scripts or steps below)

Generate the certificates according to this page https://docs.pingcap.com/tidb/stable/generate-self-signed-certificates
Modify pkg/config/config.toml.example

```
ssl-ca = "/home/genius/root.crt"
ssl-cert = "/home/genius/tikv.crt"
ssl-key = "/home/genius/tikv.key"
```

Run tidb-server:
```
./bin/tidb-server -config config.toml.example
```

Verify connection with mysql client:

```
$ mysql --tls-version=TLSv1.0 -v -v -v --ssl-ca='/home/genius/root.crt'  --ssl-cert="/home/genius/tikv.crt" --ssl-key="/home/genius/tikv.key" -uroot -h127.0.0.1  
ERROR 2026 (HY000): SSL connection error: TLS version is invalid

$ mysql --tls-version=TLSv1.1 -v -v -v --ssl-ca='/home/genius/root.crt'  --ssl-cert="/home/genius/tikv.crt" --ssl-key="/home/genius/tikv.key" -uroot -h127.0.0.1  
ERROR 2026 (HY000): SSL connection error: TLS version is invalid
```

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
remove TLS1.0, TLS1.1 support
```

### CODE REVIEW

File: br/pkg/lightning/config/config.go [hunk 1]
```diff
@@ -187,7 +187,7 @@ func (d *DBStore) adjust(
 		if d.Security.TLSConfig == nil {
 			/* #nosec G402 */
 			d.Security.TLSConfig = &tls.Config{
-				MinVersion:         tls.VersionTLS10,
+				MinVersion:         tls.VersionTLS12,
```
Comment (lance6716) [inline br/pkg/lightning/config/config.go:190]: dumpling and lightning may connect to older version of TiDB and MySQL, and their TLS version is older than v1.2. Should we support this case @frank945946

Comment (tiancaiamao) [reply br/pkg/lightning/config/config.go:190]: TLSv1.2 is introduced in year 2008, 15 years ago, that's old enough.
And as long as those software support TLSv1.2 above, change this MinVersion will not cause any trouble.
Without doubt, that's the case for TiDB.

And then I check the release history of MySQL ...
```
Release	General availability	Latest release
5.5 LTS	3 December 2010	2018-10-22
5.6 LTS	5 February 2013	2021-01-20
5.7 LTS	21 October 2015	2023-10-25
8.0 LTS	19 April 2018	2023-10-25
```

I'm sure it's pretty safe to do this change according to the above information.

**Review by bb7133 — APPROVED** [review]
LGTM

File: br/pkg/lightning/config/config.go (hunk 1 above)
Comment (bb7133) [reply br/pkg/lightning/config/config.go:190]: It is not so 'safe' actually. We do know many TiDB users still using old clients. However, it is reasonable to remove the support of 'TLS 1.0/1.1'.

Thanks @lance6716

Comment (Frank945946) [reply br/pkg/lightning/config/config.go:190]: Removing the support of 'TLS 1.0/1.1'  LTGM  and please leave a note to the corresponding docs @lance6716 .

**Review by Frank945946 — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Review by okJiang — APPROVED** [review]

**Pull request merged**